### PR TITLE
build: update CLI version to 1.7.2

### LINKED ISSUE #195: Update CLI version to latest
Update `@angular/cli` version to the latest to make easy development/contribution.

- rename `angular-cli.json` to `.angular-cli.json`
- migrate `.angular-cli.json` configurations
- add/remove some files

migrating guide: https://github.com/angular/angular-cli/wiki/stories-1.0-update

### PR DESCRIPTION
**Pull request by rafaelss95**
Closes #195.

### CODE REVIEW

**Comment by rafaelss95** [comment]
I guess `README.md` needs to be updated. `ng serve` simply doesn't work (without run another script before) (see #227). I had to run `yarn start`, in order to run the `build-themes` before start the server.

Also, there's a lint error due to `HttpModule` and its relatives (like `RequestOptions`, etc...) deprecations.
Currently there are many places using them. Is it preferable to disable this rule (for now) and replace them in another PR?

Can you take a look, @jelbourn? Thanks.

**Comment by rafaelss95** [comment]
@jelbourn In fact, there are no unrelated changes. The proposed changes are a fully update in `@angular/cli`, to be in parity with latest version. 

Look at the PR #143, the `cli` version had been updated from `"@angular/cli": "^1.0.0-beta.30"` to `"@angular/cli": "^1.0.0"`... **but** just the version was updated, completely ignoring the fact that it changed a lot from beta to the final version (file names, structure, etc).

So, what I did here was: I created a fresh new project and compared all the changes, and there are a lot of changes, as you could seen.

**Comment by rafaelss95** [comment]
Well, those tslint rules are the default in the latest `@angular/cli`. I can undo those changes if you want.

**Comment by jelbourn** [comment]
Yeah- we don't necessarily use the default tslint rules here, anyway; we'd really want the same rules as the main material2 repo (except for the custom rules, which I don't want to copy-paste).

**Comment by rafaelss95** [comment]
No problem, I undid the change of tslint.

**Comment by jelbourn** [comment]
@amcdnl could you do a review pass?

File: src/app/shared/navigation-focus/navigation-focus.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export class NavigationFocus implements OnInit {
   ngOnInit() {
     clearTimeout(lastTimeoutId);
     // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
-    lastTimeoutId = setTimeout(() => this.el.nativeElement.focus(), 100);
+    lastTimeoutId = window.setTimeout(() => this.el.nativeElement.focus(), 100);
```
Comment (amcdnl) [inline src/app/shared/navigation-focus/navigation-focus.ts]: Why the `window` here?

File: src/app/index.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
 export * from './material-docs-app';
-export * from './app-module';
+export * from './app.module';
```
Comment (amcdnl) [inline src/app/index.ts]: @jelbourn prefers the `app-module` format.

File: tsconfig.json [hunk 3]
```diff
@@ -1,16 +1,26 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
```
Comment (amcdnl) [inline tsconfig.json]: Why was this added?

**Comment by amcdnl** [comment]
@rafaelss95 - Can you rebase?

File: src/app/shared/navigation-focus/navigation-focus.ts (hunk 1 above)
Comment (rafaelss95) [reply src/app/shared/navigation-focus/navigation-focus.ts]: Not sure, but I guess I got an error when tried to run a specific command.

File: src/app/index.ts (hunk 2 above)
Comment (rafaelss95) [reply src/app/index.ts]: Really :( I'll change ASAP.

**Comment by rafaelss95** [comment]
@amcdnl I've rebased, however now there a lot of errors. Lint (which ones I fixed in this PR) and others coming from `@angular/cdk/tree` and `MatBottomSheet`.

**Comment by rafaelss95** [comment]
Done in #431.
